tool/generator/bcd.dart

```diff
@@ -37,29 +82,53 @@ class BrowserCompatData {
   BrowserCompatData(this.interfaces);
 
   BCDInterfaceStatus? retrieveInterfaceFor(String name) => interfaces[name];
+
+  bool shouldGenerateInterface(String name) =>
+      retrieveInterfaceFor(name)?.shouldGenerate ?? false;
 }
 
 class BCDInterfaceStatus extends BCDItem {
-  late final Map<String, BCDPropertyStatus> properties;
+  final Map<String, BCDPropertyStatus> _properties = {};
 
   BCDInterfaceStatus(super.name, super.json) {
-    properties = Map.fromIterable(
-      json.symbolNames,
-      value: (name) => BCDPropertyStatus(
-          name as String, json[name] as Map<String, dynamic>, this),
-    );
+    for (final symbolName in json.symbolNames) {
+      addProperty(symbolName, json[symbolName] as Map<String, dynamic>);
+    }
+  }
+
+  void addProperty(String property, Map<String, dynamic> compat) {
+    // Event compatibility data is stored as `<name_of_event>_event`. In order
+    // to have compatibility data for `onX` properties, we need to replace such
+    // property names. See https://github.com/mdn/browser-compat-data/blob/main/docs/data-guidelines/api.md#dom-events-eventname_event
+    // for more details.
+    late BCDPropertyStatus status;
+    const eventSuffix = '_event';
+    if (property.endsWith(eventSuffix)) {
+      property = 'on${property.replaceAll(eventSuffix, '')}';
```

Comment:
If I'm reading that section correctly, if an implementation (browser) doesn't support the event handler, the data uses `partial_implementation`, and if it supports the handler but not the event, the data uses `"version_added": false`.

This is however specific to a particular browser and not specific to the member itself, so I don't think it's sufficient to say whether an event handler is supported. We could check all the major browsers for this and make that determination, but that still may not tell us if a particular event handler is *actually* "experimental", because that determination is a number of factors. I think keeping it consistent with the event for now makes the most sense.

Reply:
Got it, makes sense